### ISSUE TITLE
ParallelRunStep failed in run(data) function 

### ISSUE THREAD

**Issue by lingfeidiewu** [Azure/MachineLearningNotebooks#1102]
I tried to use ParallelRunStep to do batch score. The code of step is below:
```
parallel_run_config = ParallelRunConfig(
    environment=curated_environment,
    entry_script="use_model.py",
    source_directory="./",
    output_action="append_row",
    mini_batch_size="20",
    error_threshold=1,
    compute_target=compute_target,
    process_count_per_node=2,
    node_count=2
)
datastore_path2 = [
        DataPath(def_blob_store, 'Private/Opportunities/ModelData/OneModel/usedData-FA6/AAB-I/part-00000-tid-*.snappy.parquet')
    
    ]
test_tabular = Dataset.Tabular.from_parquet_files(
        path=datastore_path2)
  
test_featuredata = test_tabular.register(workspace = ws,name='test_featuredata', create_new_version=True)
test_data = Dataset.get_by_name(ws, 'test_featuredata')
parallel_step_name = "batchscoring"
output_dir = PipelineData(name="scores", 
                          datastore=datastore,
                          )
batch_score_step1 = ParallelRunStep(
        name=parallel_step_name,
        inputs=[test_data.as_named_input("test_data")],
       # side_inputs=[output_register],
        output=output_dir,
        parallel_run_config=parallel_run_config,
        allow_reuse=False
    )
```
But when it failed at the begining of run(data) function in "use_model.py" script. The error in driver_log is below:

```
Traceback (most recent call last):
  File "driver/amlbi_main.py", line 62, in <module>
    main()
  File "driver/amlbi_main.py", line 55, in main
    booter.start()
  File "/mnt/batch/tasks/shared/LS_root/jobs/ucmopp-ws/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/mounts/workspaceblobstore/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/driver/azureml_user/parallel_run/boot.py", line 283, in start
    self.start_sys_main()
  File "/mnt/batch/tasks/shared/LS_root/jobs/ucmopp-ws/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/mounts/workspaceblobstore/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/driver/azureml_user/parallel_run/boot.py", line 199, in start_sys_main
    self.run(cmd)
  File "/mnt/batch/tasks/shared/LS_root/jobs/ucmopp-ws/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/mounts/workspaceblobstore/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/driver/azureml_user/parallel_run/boot.py", line 152, in run
    BootResult().check_result(stdout)
  File "/mnt/batch/tasks/shared/LS_root/jobs/ucmopp-ws/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/mounts/workspaceblobstore/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/driver/azureml_user/parallel_run/boot_result.py", line 51, in check_result
    self.raise_exception(error)
  File "/mnt/batch/tasks/shared/LS_root/jobs/ucmopp-ws/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/mounts/workspaceblobstore/azureml/955ad1ea-2fc6-450d-8da8-9820f4dd7ab9/driver/azureml_user/parallel_run/boot_result.py", line 38, in raise_exception
    raise Exception(exc_msg)
Exception: 'Entity' object has no attribute 'total'
```
And the error in logs/sys/error is below:
`2020-08-17 03:43:22,051|ERROR|238|Stop for run() unable to finish within 90 seconds for {"agent": "agent000", "task_id": 0, "dequeue_count": 1, "method": "run", "start_time": "2020-08-17T03:41:51.628424", "duration": 90.42170715332031, "process_time": 0.6981617629999999, "run_method_time": 90.42170715332031, "status": "RUN_TIMEOUT"}.`

What's the problem?

**Comment by tracychms** [comment]
Would you share "use_model.py"? The run() is timed out due to error: no 'total' attribute in 'Entity'. Seems you are using a non-exist attribute in run() method.

**Comment by shivanissambare** [comment]
@lingfeidiewu are you still facing this problem, if yes, can you please share what tracy has mentioned above.
if not, I will close this by end of week.
thank you :)

**Comment by amitsethitx** [comment]
I am also seeing the same problem . Any updates on this issue ?

**Comment by tracychms** [comment]
@amitsethitx when you use tabular dataset as ParallelRunStep input, the data you will get in run(mini_batch) is a locally accessible pandas dataframe. Below is a sample entry script for your reference:
https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/machine-learning-pipelines/parallel-run/Code/iris_score.py

**Comment by amitsethitx** [comment]
Hello Tracy - I tried but getting same errors.

2020-10-21T21:08:45.380600] The experiment failed. Finalizing run...
[2020-10-21T21:08:45.380667] FinalizingInRunHistory is not called
Cleaning up all outstanding Run operations, waiting 900.0 seconds
1 items cleaning up...
Cleanup took 0.0944056510925293 seconds
Traceback (most recent call last):
  File "driver/amlbi_main.py", line 70, in <module>
    main()
  File "driver/amlbi_main.py", line 63, in main
    booter.start()
  File "driver/azureml_user/parallel_run/boot.py", line 293, in start
    self.start_sys_main()
  File "driver/azureml_user/parallel_run/boot.py", line 206, in start_sys_main
    self.run(cmd)
  File "driver/azureml_user/parallel_run/boot.py", line 159, in run
    BootResult().check_result(stdout)
  File "driver/azureml_user/parallel_run/boot_result.py", line 23, in check_result
    raise ExceptionHelper.load_exc_from_json(error_json)
azureml_common.parallel_run.exception.EntryScriptException: Entry script error. No progress update in 270 seconds. Updated: False, total wait from last update: 278 seconds, remaining -8 seconds. 0/239551 items processed.
